src/websockets.ml

```diff
@@ -21,8 +21,23 @@ module Wsprotocol (IO : Iteratees.Monad) = struct
 
   type 'a t = 'a I.t 
 
-  let base64encode s = modify Base64.encode s
-  let base64decode s = modify Base64.decode s
+  let sanitize s =
+    (* ignore control characters: see RFC4648.1 and RFC4648.3
+     * https://tools.ietf.org/html/rfc4648#section-3
+     * Note: \t = \009, \n = \012, \r = \015, \s = \032 *)
```

Comment:
Hmm, I'm actually confused by what the RFC says:

>  Implementations MUST reject the encoded data if it contains
   characters outside the base alphabet when interpreting base-encoded
   data, unless the specification referring to this document explicitly
   states otherwise.  Such specifications may instead state, as MIME
   does, that characters outside the base encoding alphabet should
   simply be ignored when interpreting data ("be liberal in what you
   accept").

First it says that we MUST reject it, then it says that we MAY ignore it.